pkgs/by-name/di/digikam/package.nix

```diff
@@ -1,149 +1,213 @@
-{ stdenv, config, lib, fetchurl, cmake, doxygen, extra-cmake-modules, wrapGAppsHook3
-
-# For `digitaglinktree`
-, perl, sqlite
-
-, libsForQt5
-
-, bison
-, boost
-, eigen
-, exiv2
-, ffmpeg_4
-, flex
-, graphviz
-, imagemagick
-, lcms2
-, lensfun
-, libgphoto2
-, liblqr1
-, libusb1
-, libheif
-, libGL
-, libGLU
-, opencv
-, pcre
-, x265
-, jasper
-
-, bash
-# For panorama and focus stacking
-, enblend-enfuse
-, hugin
-, gnumake
-
-, cudaSupport ? config.cudaSupport
-, cudaPackages ? {}
+{
+  stdenv,
+  config,
+  lib,
+  fetchFromGitLab,
+  fetchgit,
+
+  cmake,
+  ninja,
+  extra-cmake-modules,
+  flex,
+  bison,
+  wrapGAppsHook3,
+
+  opencv,
+  libtiff,
+  libpng,
+  libjpeg,
+  libheif,
+  libjxl,
+  boost,
+  lcms2,
+  expat,
+  exiv2,
+  libxml2,
+  libxslt,
+  ffmpeg,
+  jasper,
+  eigen,
+  lensfun,
+  liblqr1,
+  libgphoto2,
+  libusb1,
+  imagemagick,
+  x265,
+  libGLX,
+  libGLU,
+
+  qt6Packages,
+  kdePackages,
+
+  # For `digitaglinktree`
+  perl,
+  sqlite,
+
+  runtimeShell,
+  # For panorama and focus stacking
+  enblend-enfuse,
+  hugin,
+  gnumake,
+
+  cudaSupport ? config.cudaSupport,
+  cudaPackages ? { },
 }:
 
-stdenv.mkDerivation rec {
-  pname   = "digikam";
-  version = "8.3.0";
+let
+  # Make sure the correct Darwin SDK is picked up.
+  inherit (qt6Packages.callPackage ({ stdenv }: stdenv) { }) mkDerivation;
```

Comment:
is this really required?

Reply:
Actually no, because half the stuff this depends on isn’t packaged for Darwin, sadly, so I gave up on making it work there. I can remove it.

Some explanation for the future, though:

`qt6Packages.callPackage` overrides `stdenv`. We can’t use `qt6Packages.callPackage` because this is in `pkgs/by-name`. As far as I can tell, things *happen* to work without it – you just get benign warnings about linking stuff with mismatching deployment target versions – but consider the case where a Qt 6 header starts including something from a system framework that is only present in the 11.0 SDK. In that case, the build would fail if you used the default SDK.

After the Darwin SDK rework this should simplify to `darwin.sdkMinVersion = "11";` inside the `mkDerivation` or similar, so it won’t be this gross forever, at least.

Reply:
Removed.